src/servers/Server_Zone/Script/NativeScript.cpp

```diff
@@ -0,0 +1,139 @@
+#include "NativeScript.h"
+
+namespace Core {
+    namespace Scripting {
+
+        NativeScript::NativeScript( )
+        { }
+
+        StatusEffectScript* NativeScript::getStatusEffectScript( uint32_t statusId )
+        {
+           auto script = m_statusEffectScripts.find( statusId );
+           if( script == m_statusEffectScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        ActionScript* NativeScript::getActionScript(uint32_t abilityId)
+        {
+           auto script = m_actionScripts.find( abilityId );
+           if( script == m_actionScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        QuestScript* NativeScript::getQuestScript( uint32_t questId )
+        {
+           auto script = m_questScripts.find( questId );
+           if( script == m_questScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        BattleNpcScript* NativeScript::getBattleNpcScript( uint32_t npcId )
+        {
+           auto script = m_battleNpcScripts.find( npcId );
+           if( script == m_battleNpcScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        ZoneScript* NativeScript::getZoneScript( uint32_t zoneId )
+        {
+           auto script = m_zoneScripts.find( zoneId );
+           if( script == m_zoneScripts.end() )
+              return nullptr;
+
+           return script->second;
+        }
+
+        bool NativeScript::loadScript( std::string path )
```

Comment:
any reason why this isn't pass by reference + const? from what I can tell you don't need mutability for path

Reply:
not really lol, fixed too